version: '3.9' 

services:
  django:
    build:
      context: ./django 
    container_name: django_app 
    command: python manage.py runserver 0.0.0.0:8000 
    volumes:
      - ./django:/app 
    env_file:
      - .env.test
    ports:
      - "8000:8000" 
    depends_on:
      db:
        condition: service_healthy
    healthcheck:
      test: ["CMD-SHELL", "curl -f http://localhost:8000/ || exit 1"]
      interval: 3s
      timeout: 1s
      retries: 10

  db:
    image: postgres:17-alpine 
    container_name: postgres_db 
    env_file:
      - .env.test
    volumes:
      - postgres_data:/var/lib/postgresql/data 
    ports:
      - "5432:5432" 
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 3s
      timeout: 1s
      retries: 10
      

  nginx:
    image: nginx:mainline-alpine 
    container_name: nginx 
    ports:
      - "80:80"
    volumes:
      - ./nginx:/etc/nginx/conf.d 
    depends_on:
      django:
        condition: service_healthy 

volumes:
  postgres_data: 
